Dockerfile

```diff
@@ -54,49 +52,38 @@ ARG DALI_TIMESTAMP
 # Optional build arguments
 
 ARG CMAKE_BUILD_TYPE=Release
+ENV CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
 ARG BUILD_TEST=ON
+ENV BUILD_TEST=${BUILD_TEST}
 ARG BUILD_BENCHMARK=ON
+ENV BUILD_BENCHMARK=${BUILD_BENCHMARK}
 ARG BUILD_NVTX=OFF
+ENV BUILD_NVTX=${BUILD_NVTX}
 ARG BUILD_PYTHON=ON
+ENV BUILD_PYTHON=${BUILD_PYTHON}
 ARG BUILD_LMDB=ON
+ENV BUILD_LMDB=${BUILD_LMDB}
 ARG BUILD_TENSORFLOW=ON
+ENV BUILD_TENSORFLOW=${BUILD_TENSORFLOW}
 ARG BUILD_JPEG_TURBO=ON
+ENV BUILD_JPEG_TURBO=${BUILD_JPEG_TURBO}
 ARG BUILD_NVJPEG=ON
+ENV BUILD_NVJPEG=${BUILD_NVJPEG}
 ARG BUILD_NVOF=ON
+ENV BUILD_NVOF=${BUILD_NVOF}
 ARG BUILD_NVDEC=ON
+ENV BUILD_NVDEC=${BUILD_NVDEC}
 ARG BUILD_NVML=ON
-
-RUN LD_LIBRARY_PATH="${PWD}:${LD_LIBRARY_PATH}" && \
-    cmake ../ -DCMAKE_INSTALL_PREFIX=. -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} \
-          -DBUILD_TEST=${BUILD_TEST} -DBUILD_BENCHMARK=${BUILD_BENCHMARK} \
-          -DBUILD_NVTX=${BUILD_NVTX} -DBUILD_PYTHON=${BUILD_PYTHON} \
-          -DBUILD_LMDB=${BUILD_LMDB} -DBUILD_TENSORFLOW=${BUILD_TENSORFLOW}  \
-          -DBUILD_JPEG_TURBO=${BUILD_JPEG_TURBO} -DBUILD_NVJPEG=${BUILD_NVJPEG} \
-          -DBUILD_NVOF=${BUILD_NVOF} -DBUILD_NVDEC=${BUILD_NVDEC} \
-          -DBUILD_NVML=${BUILD_NVML} \
-          -DWERROR=ON \
-          -DDALI_BUILD_FLAVOR=${NVIDIA_DALI_BUILD_FLAVOR} \
-          -DTIMESTAMP=${DALI_TIMESTAMP} -DGIT_SHA=${GIT_SHA} && \
-    make -j"$(grep ^processor /proc/cpuinfo | wc -l)"
+ENV BUILD_NVML=${BUILD_NVML}
 
 ARG NVIDIA_BUILD_ID
 ENV NVIDIA_BUILD_ID ${NVIDIA_BUILD_ID:-0}
+RUN mkdir /wheelhouse && chmod 0777 /wheelhouse
+
+FROM builder
+COPY . .
```

Comment:
so it still copies the whole source tree to the docker image. Right?

Reply:
Yes. But you can create `builder` which doesn't (is subimage) and full image that works as it used to be.